sopel/tools/jobs.py

```diff
@@ -105,47 +120,34 @@ def run(self):
                 # the log with useless error messages.
                 time.sleep(10.0)  # seconds
 
-    def _do_next_job(self):
-        """Wait until there is a job and do it."""
+    def _get_ready_jobs(self, now):
         with self._mutex:
-            # Wait until the next job should be executed.
-            # This has to be a loop, because signals stop time.sleep().
-            while True:
-                job = self._jobs.peek()
-                difference = job.next_time - time.time()
-                duration = min(difference, self.min_reaction_time)
-                if duration <= 0:
-                    break
-                with released(self._mutex):
-                    time.sleep(duration)
-
-            self._cleared = False
-            job = self._jobs.get()
-            with released(self._mutex):
-                if job.func.thread:
-                    t = threading.Thread(
-                        target=self._call, args=(job.func,)
-                    )
-                    t.start()
-                else:
-                    self._call(job.func)
-                job.next()
-            # If jobs were cleared during the call, don't put an old job
-            # into the new job queue.
-            if not self._cleared:
-                self._jobs.put(job)
+            jobs = [job for job in self._jobs if job.is_ready_to_run(now)]
+
+        return jobs
+
+    def _run_job(self, job):
+        if job.func.thread:
+            t = threading.Thread(
+                target=self._call, args=(job.func,)
+            )
+            t.start()
+        else:
+            self._call(job.func)
+        job.next()
 
     def _call(self, func):
         """Wrapper for collecting errors from modules."""
-        # Sopel.bot.call is way too specialized to be used instead.
         try:
             func(self.bot)
+        except KeyboardInterrupt:
+            # Do not lock on KeyboardInterrupt
```

Comment:
(Same as above one.)
```suggestion
            # Do not block on KeyboardInterrupt
```